Fix possible bad boot filename:

jacobweinstock opened the pull request:
## Description

<!--- Please describe what this PR is going to change -->
If no opentelemetry traceparent is found in the context while setting the bootfileAndNextServer but otel is enabled we were sending a bootfile name like: `unidonly-`. This fixes this.

## Why is this needed

<!--- Link to issue you have raised -->

Fixes: #

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->


## How are existing users impacted? What migration steps/scripts do we need?

<!--- Fixes a bug, unblocks installation, removes a component of the stack etc -->
<!--- Requires a DB migration script, etc. -->


## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

displague commented:
Let's add a test for this change. We have an existing block: https://github.com/tinkerbell/dhcp/blob/main/option_test.go#L147-L237

jacobweinstock committed:
Fix possible bad boot filename:

If no traceparent is found in the context but otel
is enabled we were sending a bootfile name like: `unidonly-`.

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

jacobweinstock commented:
> Let's add a test for this change. We have an existing block: https://github.com/tinkerbell/dhcp/blob/main/option_test.go#L147-L237

sure thing. test added.

jacobweinstock commented:
Hey @displague. When/If you have any cycles, would you mind checking this out?

jacobweinstock commented:
Hey @nshalman, if/when you have some cycles would you be able to review this, please?

nshalman reviewed (approved):
Looks good.
